tlatools/org.lamport.tlatools/src/tlc2/value/ValueInputStream.java

```diff
@@ -91,7 +91,27 @@ public final IValue read() throws IOException {
 		}
 	}
 	
+	/**
+	 * This used to be readExternal, but the tbl parameter did not end up being useful.
+	 * It was meant to allow using a different string intern table, but as a result
+	 * required all deserialized string values to be known ahead of time (in the map).
+	 * 
+	 * This stub calls readExternal, ignoring tbl and directly loading new strings into
+	 * this TLC's global intern table instead. It is left here to avoid Java method
+	 * errors with community modules, which call this signature in particular.
+	 * 
+	 * TODO:
+	 * - Once all practically deployed TLCs that don't have readExternal are phased out,
+	 *   change community modules to call readExternal.
+	 * - Once all practically deployed community modules that use this method are phased out,
+	 *   delete it.
+	 */
+	@Deprecated
 	public final IValue read(final Map<String, UniqueString> tbl) throws IOException {
+		return readExternal();
+	}
+	
+	public final IValue readExternal() throws IOException {

```

Comment:
I would add a docstring to this method about the name and purpose of the method; something like
```java
/**
 * Safely read a value from an external source that may have a different intern table for UniqueStrings.
 * By contrast, {@link #read()} assumes that the value was written by this process using the same intern table.
 */
```

(Otherwise this change looks great to me! 👍 )

Reply:
Good point. I liked your phrasing, so I used it verbatim.